CMakeLists.txt

```diff
@@ -292,10 +292,20 @@ if(WITH_TESTS)
   list(APPEND BOOST_COMPONENTS unit_test_framework)
 endif()
 
-find_package(Boost REQUIRED ${BOOST_COMPONENTS})
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  set(BOOST_MINIMUM_VERSION "1.65.0")
+else()
+  set(BOOST_MINIMUM_VERSION "1.53.0")
+endif()
+
+find_package(Boost ${BOOST_MINIMUM_VERSION} REQUIRED ${BOOST_COMPONENTS})
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 list(APPEND LIBRARIES ${Boost_LIBRARIES})
 
+if(Boost_VERSION VERSION_GREATER 106399 AND Boost_VERSION VERSION_LESS 106500)
+  message(FATAL_ERROR "Boost 1.64 is unsupported")
```

Comment:
Why is Boost 1.64 not supported? It's currently the standard version in fedora...

Reply:
Its Boost.MPI is incompatible with its Boost.Serialization. Boost 1.64 simply cannot be compiled if MPI is enabled. There is no way to make it work besides upgrading or downgrading. See https://svn.boost.org/trac10/ticket/12723 .

Reply:
Removing these lines, I can compile on my fedora 27, with boost version 1.64.
All test run successfully (in parallel), except for `testsuite/thermalized_bond.py`, which only succeeds when run single-core.

Reply:
Then Fedora must have backported that patch but left the Boost version unchanged. Can you check Fedora's source package repository for that? I guess we would need some way to detect whether we are using a vanilla Boost 1.64 or a patched one if we want to warn the user that their Boost version may be broken. If we don't do that, they will just get hard-to-understand error messages.